internal/test/daemon/daemon_unix.go

```diff
@@ -20,13 +21,29 @@ func cleanupNetworkNamespace(t testingT, execRoot string) {
 	// cleaned up when a new daemon is instantiated with a
 	// new exec root.
 	netnsPath := filepath.Join(execRoot, "netns")
-	filepath.Walk(netnsPath, func(path string, info os.FileInfo, err error) error {
-		if err := unix.Unmount(path, unix.MNT_FORCE); err != nil {
+
+	files, err := ioutil.ReadDir(netnsPath)
+	if err != nil {
+		t.Logf("unable to read netns path (%s): %s", netnsPath, err)
+		return
+	}
+
+	for _, d := range files {
+		if !d.IsDir() {
+			continue
+		}
+
+		name := d.Name()
+		path := filepath.Join(netnsPath, name)
+
+		// Unmount the network namespace using MNT_DETACH (lazy unmount).
+		// Ignore EINVAL ("target is not a mount point"), and ENOENT (not found),
+		// which could occur if the network namespace was already gone.
+		if err := unix.Unmount(path, unix.MNT_DETACH); err != nil && err != unix.EINVAL && err != unix.ENOENT {
```

Comment:
Ah, yes, rebase is needed due to https://github.com/moby/moby/commit/73baee2dcf546b2561bdd9a500b0af08cb62b1be (https://github.com/moby/moby/pull/38127), which largely did the same

I think this PR still has merit, as it removes the use of `filepath.Walk`; let me rebase